### PR TITLE
Adjust to AGP >=7.3.0 behavior change, fix build failures on empty sourceSets

### PR DESCRIPTION
**Pull request by mateuszkwiecinski**
Fixes https://github.com/jeremymailen/kotlinter-gradle/issues/253

### CODE REVIEW

File: src/main/kotlin/org/jmailen/gradle/kotlinter/pluginapplier/AndroidSourceSetApplier.kt [hunk 1]
```diff
@@ -28,8 +27,10 @@ internal object AndroidSourceSetApplier : SourceSetApplier {
             .getOrNull()
             .orEmpty()
 
+        val emptyFileTree = project.files().asFileTree
+
         return (javaSources + kotlinSources)
             .map { dir -> project.fileTree(dir) { it.include("**/*.kt") } }
-            .reduce { merged: FileTree, tree: ConfigurableFileTree -> merged + tree }
+            .fold(emptyFileTree) { merged, tree -> merged + tree }
```
Comment (mateuszkwiecinski) [inline src/main/kotlin/org/jmailen/gradle/kotlinter/pluginapplier/AndroidSourceSetApplier.kt:34]: So, it seems like AGP starting from 7.3.0 doesn't include empty directories in their source sets. On 7.2.1, `javaSources` contains `src/main/java`, `kotlinSources` holds a reference to `src/main/kotlin`.
The behavior I observed on 7.3.0 was empty `javaSources` and `kotlinSources` if they did not contain any files. 
To reproduce the issue, it's enough to bump `Versions.androidTools` to anything above `7.2.1` and run existing test suite

File: src/main/kotlin/org/jmailen/gradle/kotlinter/pluginapplier/AndroidSourceSetApplier.kt [hunk 2]
```diff
@@ -28,8 +27,10 @@ internal object AndroidSourceSetApplier : SourceSetApplier {
             .getOrNull()
             .orEmpty()
 
+        val emptyFileTree = project.files().asFileTree
```
Comment (mateuszkwiecinski) [inline src/main/kotlin/org/jmailen/gradle/kotlinter/pluginapplier/AndroidSourceSetApplier.kt:30]: About the fix itself: `reduce` can't operate on empty collections, so I replaced it with `fold` and provide the initial value. 
The way I obtain initial value is inspired by this thread: https://discuss.gradle.org/t/re-how-to-create-an-empty-filetree/14362

File: src/main/kotlin/org/jmailen/gradle/kotlinter/KotlinterPlugin.kt [hunk 3]
```diff
@@ -42,8 +42,8 @@ class KotlinterPlugin : Plugin<Project> {
                         lintTask.ignoreFailures.set(provider { kotlinterExtension.ignoreFailures })
                         lintTask.reports.set(
                             provider {
-                                kotlinterExtension.reporters.associate { reporter ->
-                                    reporter to reportFile("$id-lint.${reporterFileExtension(reporter)}")
+                                kotlinterExtension.reporters.associateWith { reporter ->
```
Comment (mateuszkwiecinski) [inline src/main/kotlin/org/jmailen/gradle/kotlinter/KotlinterPlugin.kt:45]: image

I navigate through IDE via `F2` (next highlighted error) so I usually try to fix things like this 😄

**Review by jeremymailen — APPROVED** [review]
Looks great! Feel free to ship a patch release as well to unblock folks.

**Pull request merged**